src/uu/fold/src/fold.rs

```diff
@@ -10,58 +10,82 @@
 #[macro_use]
 extern crate uucore;
 
+use clap::{App, Arg};
 use std::fs::File;
 use std::io::{stdin, BufRead, BufReader, Read};
 use std::path::Path;
 
+static NAME: &str = "fold";
+static VERSION: &str = env!("CARGO_PKG_VERSION");
 static SYNTAX: &str = "[OPTION]... [FILE]...";
 static SUMMARY: &str = "Writes each file (or standard input if no files are given)
  to standard output whilst breaking long lines";
-static LONG_HELP: &str = "";
+
+mod options {
+    pub const BYTES: &str = "bytes";
+    pub const SPACES: &str = "spaces";
+    pub const WIDTH: &str = "width";
+    pub const FILE: &str = "file";
+}
 
 pub fn uumain(args: impl uucore::Args) -> i32 {
     let args = args.collect_str();
 
     let (args, obs_width) = handle_obsolete(&args[..]);
-    let matches = app!(SYNTAX, SUMMARY, LONG_HELP)
-        .optflag(
-            "b",
-            "bytes",
-            "count using bytes rather than columns (meaning control characters \
+    let matches = App::new(executable!())
+        .name(NAME)
+        .version(VERSION)
+        .usage(SYNTAX)
+        .about(SUMMARY)
+        .arg(
+            Arg::with_name(options::BYTES)
+                .long(options::BYTES)
+                .short("b")
+                .help(
+                    "count using bytes rather than columns (meaning control characters \
              such as newline are not treated specially)",
+                )
+                .takes_value(false),
         )
-        .optflag(
-            "s",
-            "spaces",
-            "break lines at word boundaries rather than a hard cut-off",
+        .arg(
+            Arg::with_name(options::SPACES)
+                .long(options::SPACES)
+                .short("s")
+                .help("break lines at word boundaries rather than a hard cut-off")
+                .takes_value(false),
         )
-        .optopt(
-            "w",
-            "width",
-            "set WIDTH as the maximum line width rather than 80",
-            "WIDTH",
+        .arg(
+            Arg::with_name(options::WIDTH)
+                .long(options::WIDTH)
+                .short("w")
+                .help("set WIDTH as the maximum line width rather than 80")
+                .value_name("WIDTH")
+                .allow_hyphen_values(true)
+                .takes_value(true),
         )
-        .parse(args);
-
-    let bytes = matches.opt_present("b");
-    let spaces = matches.opt_present("s");
-    let poss_width = if matches.opt_present("w") {
-        matches.opt_str("w")
-    } else {
-        obs_width
+        .arg(Arg::with_name(options::FILE).hidden(true).multiple(true))
+        .get_matches_from(args.clone());
+
+    let bytes = matches.is_present(options::BYTES);
+    let spaces = matches.is_present(options::SPACES);
+    let poss_width = match matches.is_present(options::WIDTH) {
+        true => matches.value_of(options::WIDTH).map(|v| v.to_owned()),
+        false => obs_width,
```

Comment:
Could be done like this, which seems slightly more readable:
```
let poss_width = match matches.value_of(options::WIDTH) {
    Some(v) => Some(v.to_owned()),
    None => obs_width,
};
```

Reply:
sure, updated :)